### PR TITLE
Emmelia ara v3 azure backup changes hardwaresecuritymodules microsoft.hardware security modules 2024-06-30-preview

### PR DESCRIPTION
**Pull request by emmeliaAra**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

spec_pr_review_workflow_diagram

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been created in adherence to [OpenAPI specs PR creation guidance](https://aka.ms/azsdkdocs/createopenapispec)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://aka.ms/azurerpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- If you don't have permissions to remove or add labels to the PR, request `write access` per [aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories](https://aka.ms/azsdk/access#request-access-to-rest-api-or-sdk-repositories")
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with ARM review (PR workflow diagram Step 2), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

### CODE REVIEW

**Comment by emmeliaAra** [comment]
Hi, @mikekistler 
Can you please help with the breaking change on this PR. The changes are already approved in another PR after joining office hours and we need to duplicate the spec version for a preview version. 

Already approved PR link: https://github.com/Azure/azure-rest-api-specs/pull/28651

**Review by mikekistler — DISMISSED** [review]
The Azure Versioning policy does not allow a GA and a preview API version with the same date. Since PR #28651 is adding this 2024-06-30 GA API version, the new preview must have an API version that is 2024-07-01-preview or later.

Furthermore, you need to merge #28651, and then rerun the pipeline checks before this PR is merged. That should clear any breaking changes issues and if not we'll need to understand why.

**Comment by mikekistler** [comment]
Please come to [Breaking Change office hours](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/apex_support) on Monday at 10AM PT, or schedule a separate meeting with me if that's not convenient, so we can talk through this.

**Comment by emmeliaAra** [comment]
> Please come to [Breaking Change office hours](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/apex_support) on Monday at 10AM PT, or schedule a separate meeting with me if that's not convenient, so we can talk through this.

will do, thanks!

**Comment by raosuhas** [comment]
Can you please fix the LintDiff warnings introduced in this PR : https://github.com/Azure/azure-rest-api-specs/pull/29712/checks?check_run_id=27324030518

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json
Comment (raosuhas) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json:1397]: x-ms-secret please

Comment (raosuhas) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json:1397]: [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-secret)

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/examples/DedicatedHsm_Update.json
Comment (raosuhas) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/examples/DedicatedHsm_Update.json:50]: >/some.endpoint.com [](http://example.com/codeflow?start=28&length=18)

nit: suggest using consistent endpoints in your examples. Management.azure.com for example

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/dedicatedhsm.json
Comment (raosuhas) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/dedicatedhsm.json:423]: format : armid  with details : 

[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details)

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/dedicatedhsm.json
Comment (raosuhas) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/dedicatedhsm.json:411]: add details if applicable : 
[https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details)

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json
Comment (raosuhas) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json:1409]: just to be clear , the secret part of the sas uri is not present here right ?

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json
Comment (raosuhas) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json:1191]: use enum. Please review the whole PR and check if objects are better modelled as enums rather than free form strings

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json
Comment (raosuhas) [inline specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json:674]: >backup [](http://example.com/codeflow?start=155&length=6)

Please keep in mind that by choosing to model these as POSTS , you will not be able to trigger these in ARM templates or invoke these as part of ARM policy affects . Is this acceptable for your scenarios ?

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json
Comment (emmeliaAra) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json:1409]: yess, we do not return the SAS token in the result, just the blob storage account and the container: "https://myaccount.blob.core.windows.net/container/container",

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json
Comment (emmeliaAra) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json]: Updated string properties to Enums where applicable.

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json
Comment (emmeliaAra) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json:674]: Hi @raosuhas, what is  ARM policy affect?

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/examples/DedicatedHsm_Update.json
Comment (emmeliaAra) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/examples/DedicatedHsm_Update.json]: updated endpoints in all examples

File: specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/dedicatedhsm.json
Comment (emmeliaAra) [reply specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/dedicatedhsm.json:424]: updated

**Comment by emmeliaAra** [comment]
> Can you please fix the LintDiff warnings introduced in this PR : https://github.com/Azure/azure-rest-api-specs/pull/29712/checks?check_run_id=27324030518

Hi @raosuhas, resolved the LintDiff warnings were possible. Please have a look at the ones below:

For [**LatestVersionOfCommonTypesMustBeUsed**](https://github.com/Azure/azure-openapi-validator/blob/main/docs/latest-version-of-common-types-must-be-used.md): I am using v6 and it is saying to use latest v5. Shall I downgrade or is it a false positive ?

For [**PatchSkuProperty**](https://github.com/Azure/azure-openapi-validator/blob/main/docs/patch-sku-property.md): Our service does not support Patch update. I added x-ms-mutability[read, create] in SKU definition ([link](https://github.com/Azure/azure-rest-api-specs/blob/278511369ed7645542eae68523246bd9bbbe980b/specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/preview/2024-06-30-preview/cloudhsm.json#L1110)) 

For [**EnumInsteadOfBoolean**](https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md): This value used to be an enum but it was confusing to customers. It can only have 2 values. ApprovedFips or NotApprovedFips so we made it a Boolean instead. 

For [**PageableOperation**](https://github.com/Azure/azure-openapi-validator/blob/main/docs/pageable-operation.md): the get operation that the warning is flagged on, it is just a GET on a specific resource and it will only return max one dedicated HSM resource and never a list.

Thanks!

**Comment by emmeliaAra** [comment]
@raosuhas gentle ping. Made some changes based on the feedback, can you have a look on the updates?

**Review by D-Alegria — APPROVED** [review]

**Review by seanmorrissey98 — APPROVED** [review]

**Pull request merged**
